src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets

```diff
@@ -115,27 +116,35 @@
           DependsOnTargets="Protobuf_Compile"
           Condition=" '$(Language)' == 'C#' " />
 
-  <Target Name="_Protobuf_SelectFiles">
+  <Target Name="_Protobuf_SetProtoRoot">
     <!-- Guess .proto root for the files. Whenever the root is set for a file explicitly,
          leave it as is. Otherwise, for files under the project directory, set the root
          to "." for the project's directory, as it is the current when compiling; for the
-         files outside of project directory, use each .proto file's directory as the root. -->
+         files outside of project directory, use each .proto file's directory as the root 
+		 or Protobuf_ProtoRoot if set. -->
     <FindUnderPath Path="$(MSBuildProjectDirectory)"
                    Files="@(Protobuf->WithMetadataValue('ProtoRoot',''))">
       <Output TaskParameter="InPath" ItemName="_Protobuf_NoRootInProject"/>
       <Output TaskParameter="OutOfPath" ItemName="_Protobuf_NoRootElsewhere"/>
     </FindUnderPath>
+	
+    <ItemGroup>
+      <Protobuf>
```

Comment:
how do we know that this change won't break existing users?

Reply:
It's the same logic but on `Protobuf` instead of `Protobuf_Compile` as we have to patch it for all items iterated over in `Protobuf_Clean`, not just the one that should be compiled. I only also added the logic for `$(Protobuf_ProtoRoot)` in additionally. So if you prefer, I can split it into more PRs. One for fixing the C++ bug, one for adding `$(Protobuf_ProtoRoot)` and keep this one only for the `Avoid collisions` part and have it blocked by the other 2 PRs.